app/components/submission/ManuscriptMetadata/SubjectAreaDropdown.test.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+import theme from '@pubsweet/elife-theme'
+
+import SubjectAreaDropdown from './SubjectAreaDropdown'
+
+const onChange = jest.fn()
+
+const MyDropdown = (
+  <SubjectAreaDropdown
+    label="My label"
+    name="My name"
+    onBlur={jest.fn()}
+    onChange={onChange}
+    savedValues={[]}
+    theme={theme}
+  />
+)
+
+describe('SubjectAreaDropdown component', () => {
+  it('renders label & select', () => {
+    const wrapper = shallow(MyDropdown).dive() // dive is necessary because SubjectAreaDropdown is wrapped by withTheme
+    expect(wrapper.contains('My label')).toBe(true)
+    expect(wrapper.find('[inputId="subject-area-select"]').exists()).toEqual(
+      true,
+    )
+    expect(wrapper.contains('No more than two subject areas')).toBe(false)
+  })
+
+  describe('Integration with react-select', () => {
+    let wrapper, selectWrapper, selectInput
+
+    beforeEach(() => {
+      wrapper = mount(MyDropdown)
+      selectWrapper = wrapper.find('Select')
+      selectInput = selectWrapper.find('input')
+    })
+
+    const typeSubjectArea = letters => {
+      // We need 2 steps to programmatically simulate typing in a letter
+      // 1. Add a letter to the value of the input field (does not cause the dropdown to appear by itself)
+      selectWrapper.props().onInputChange(letters)
+      // 2. Open the dropdown
+      selectInput.simulate('change')
+    }
+
+    it('focusing causes the dropdown to open & display every option', () => {
+      typeSubjectArea('') // typing nothing into the field seems to be the only way to programmatically focus react-select)
+      expect(wrapper.find('Menu').exists()).toEqual(true)
+      expect(wrapper.find('Option')).toHaveLength(17)
+    })
+
+    it('an option can be selected from the dropdown', () => {
+      typeSubjectArea('b')
+      selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
+      expect(onChange.mock.calls).toHaveLength(1)
+      expect(wrapper.find('Menu').exists()).toEqual(false)
+      expect(wrapper.find('MultiValue')).toHaveLength(1) // MultiValue = a "tag" (i.e. selected option) in react-select
+      expect(
+        wrapper
+          .find('MultiValue')
+          .contains('Biochemistry and Chemical Biology'),
+      ).toBe(true)
+    })
+
+    it('2 options can be selected from the dropdown', () => {
+      typeSubjectArea('b')
+      selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
+      typeSubjectArea('c')
+      selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
+      expect(wrapper.find('MultiValue')).toHaveLength(2)
+      expect(wrapper.find('MultiValue').contains('Cancer Biology')).toBe(true)
```

Comment:
Hm, this seems a bit random, you only check for `Cancer Biology` and not `Biochemistry and Chemical Biology`. You hate `Biochemistry and Chemical Biology` that much?

Reply:
I check whether it contains `Biochemistry...` [when I'm testing whether 'an option can be selected from the dropdown'](https://github.com/elifesciences/elife-xpub/blob/c8bfffb3442edb859a5101bec59824006882d48b/app/components/submission/ManuscriptMetadata/SubjectAreaDropdown.test.js#L59-L63)

I guess I figured that I only needed to check that the 2nd one had been added, without asserting that `Biology...` was still there (since I assert that it's there just above).

I figured it was safe to assume that selecting a second option wasn't going to cause the first tag to disappear. Do you think that's an unsafe assumption?

Reply:
I usually try to write unit tests as independent as possible so I would probably test for the second one too without assuming stuff. Again, open discussion :)

Reply:
Yeah I totally get where you're coming from. But these aren't really unit tests.

I think I've introduced confusion by separating it into so many little blocks in the first place. I originally had most of this in a single `it` block, but it just felt so weird to be writing one giant test for the whole component. My thinking was that, if this whole test fails, we won't know why. But I guess that comes back to the reason why you write unit tests in the first place.

Thinking about it again, it actually doesn't really add any value to separate out the tests that require this code to be repeated:
```
typeSubjectArea('b')
selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
typeSubjectArea('c')
selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
```

I'm going to suggest that I stick the first 5 tests back into a single `it` block? But it sounds like we really do need a broader discussion on whether to add unit tests (e.g. snapshot testing)

Reply:
I think I'm going to remove this test altogether. It's not really adding anything, that isn't already tested within this test: `an option can be selected from the dropdown`. If `react-select` is working as it should, then adding 1 & adding 2 are the same thing.